packages/ove-ui-launcher/src/steps/Confirm.jsx

```diff
@@ -9,91 +9,83 @@ import 'codemirror/mode/shell/shell';
 import 'codemirror/lib/codemirror.css';
 import 'codemirror/theme/dracula.css';
 
-// Setup jQuery to work inside React
-import $ from 'jquery';
-window.$ = $;
+import { Form, Button, Divider, Header } from 'semantic-ui-react';
 
 export default class Confirm extends Component {
     constructor (props) {
         super(props);
 
-        this.log = props.getLogger();
-
         this.state = {
-            app: props.getStore().app,
-            space: props.getStore().space,
-            deleteSections: props.getStore().deleteSections,
-            showController: props.getStore().showController,
-            payload: {
-                space: props.getStore().space,
-                ...props.getStore().geometry,
-                app: {
-                    url: 'http://' + Constants.REACT_APP_OVE_APP(props.getStore().app),
-                    states: {}
-                }
-            },
-            os: Constants.OS.UNIX
+            payload: { }
         };
 
-        if ([Constants.App.REPLICATOR, Constants.App.CONTROLLER].includes(this.state.app)) {
-            this.state.payload.app.states.load = JSON.parse(props.getStore().config);
-        } else if ([Constants.App.ALIGNMENT, Constants.App.WHITEBOARD].includes(this.state.app)) {
-            delete this.state.payload.app.states;
-        } else if ([Constants.App.WEBRTC].includes(this.state.app) || props.getStore().mode !== Constants.Mode.NEW) {
-            this.state.payload.app.states.load = props.getStore().state;
-        } else {
-            this.state.payload.app.states.load = props.getStore().config ? JSON.parse(props.getStore().config) : {};
-        }
-        this.log.debug('Computed payload for create section operation:', this.state.payload);
+        this.constructPayload = this.constructPayload.bind(this);
+        this._launch = this._launch.bind(this);
+    }
+
+    componentDidMount () {
+        this.constructPayload(this.props); // can't call from constructor, as setState can't be called on a component until it is mounted
+    }
 
-        this.changeSelect = this.changeSelect.bind(this);
-        this.isValidated = this.isValidated.bind(this);
-        this.log.debug('Displaying step:', Confirm.name);
+    componentDidUpdate (prevProps, prevState, snapshot) {
+        if (this.props !== prevProps) {
```

Comment:
Did you mean property compare rather than instance compare ? In the current form the render will be always triggered.

Reply:
I've realised I can instead make this whole component a stateless function, and not use any of the react lifecycle methods.